src/screens/OtherProfile/hooks/useOtherProfileScreenHooks.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+
+import StorageUtils from '../../../utils/storage';
+import {Context} from '../../../context';
+import {checkUserBlock, getOtherProfile} from '../../../service/profile';
+import {setOtherProfileFeed} from '../../../context/actions/otherProfileFeed';
+
+const useOtherProfileScreenHooks = (targetUserProfileId: string, username: string) => {
+  /**
+   * State
+   */
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [isProfileFetching, setIsProfileFetching] = React.useState<boolean>(true);
+  const [isBlocked, setIsBlocked] = React.useState<boolean | undefined>(undefined);
+  const [isBlocking, setIsBlocking] = React.useState<boolean | undefined>(undefined);
+  const [otherProfileData, setOtherProfileData] = React.useState<any>({});
+  const [, setOtherProfileUserId] = React.useState<string>('');
+
+  /**
+   * Other Profile Context
+   */
+  const [otherProfileFeeds, dispatchOtherProfile] = (React.useContext(Context) as any)
+    ?.otherProfileFeed;
+  const {feeds} = otherProfileFeeds;
+
+  const getCache = async () => {
+    const isBlockedFromCache = StorageUtils.blockedStatus.getForKey(targetUserProfileId) === 'true';
+    const isBlockingFromCache =
+      StorageUtils.blockingStatus.getForKey(targetUserProfileId) === 'true';
+
+    try {
+      const otherProfileDataCache =
+        StorageUtils.otherProfileData.getForKey(targetUserProfileId) || '{}';
+      setOtherProfileData(JSON.parse(otherProfileDataCache));
+      if (otherProfileDataCache !== '{}') {
+        setIsProfileFetching(false);
+      }
+    } catch (e) {
+      console.log('error parsing JSON other profile data', e);
+    }
+
+    try {
+      const feedsCache = StorageUtils.otherProfileFeed.getForKey(targetUserProfileId) || '[]';
+
+      setOtherProfileFeed(JSON.parse(feedsCache), dispatchOtherProfile);
+    } catch (e) {
+      console.log('error parsing JSON other profile feed', e);
+    }
+
+    setIsBlocked(isBlockedFromCache);
+    setIsBlocking(isBlockingFromCache);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a potential issue with the way you're handling cache. You're assuming that if the value for `isBlockedFromCache` and `isBlockingFromCache` is 'true', then it's true, otherwise it's false. This might not be accurate if the values are not set or if they're set to something other than 'true'. It would be better to explicitly check for both 'true' and 'false' values.

```diff
- const isBlockedFromCache = StorageUtils.blockedStatus.getForKey(targetUserProfileId) === 'true';
- const isBlockingFromCache = StorageUtils.blockingStatus.getForKey(targetUserProfileId) === 'true';
+ const isBlockedFromCache = StorageUtils.blockedStatus.getForKey(targetUserProfileId);
+ const isBlockingFromCache = StorageUtils.blockingStatus.getForKey(targetUserProfileId);
+ setIsBlocked(isBlockedFromCache === 'true');
+ setIsBlocking(isBlockingFromCache === 'true');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->